runtime/onert/backend/trix/ops/BulkLayer.cc

```diff
@@ -49,24 +50,57 @@ void BulkLayer::configure(const std::vector<const IPortableTensor *> &inputs,
     throw std::runtime_error("Unable to extract the model metadata");
   }
 
+  _model_id.resize(_dev_context->getDevSize());
+
   generic_buffer model_file;
   model_file.type = BUFFER_FILE;
   model_file.filepath = binary_path.c_str();
   model_file.size = _meta->size;
 
-  if (registerNPUmodel(dev_context->getDev(), &model_file, &_model_id) < 0)
+  if (registerNPUmodel(dev_context->getDev(0), &model_file, &_model_id[0]) < 0)
+  {
+    throw std::runtime_error("Failed to register npu model");
+  }
+  if (registerNPUmodel(dev_context->getDev(1), &model_file, &_model_id[1]) < 0)
   {
     throw std::runtime_error("Failed to register npu model");
   }
 }
 
+void single_job(npudev_h dev, int req_id, input_buffers *input_buf, tensors_data_info *in_info,
+                output_buffers *output_buf, tensors_data_info *out_info)
+{
+  if (setNPU_requestData(dev, req_id, input_buf, in_info, output_buf, out_info))
+  {
+    throw std::runtime_error("Unable to create NPU request for red_id (" + std::to_string(req_id) +
+                             ")");
+  }
+
+  if (submitNPU_request(dev, req_id))
+  {
+    throw std::runtime_error("Unable to submit NPU request with req id (" + std::to_string(req_id) +
+                             ")");
+  }
+}
+
 void BulkLayer::run()
 {
-  int req_id;
-  if (createNPU_request(_dev_context->getDev(), _model_id, &req_id))
+  // TODO: Remove too many assumption
+  // We assume user wants batch execurion if user's input size is larger than model's input size
+  int user_input_batch = (_inputs[0]->get_info().shape()).dim(0);
+  int model_input_batch = _meta->input_seg_dims[0][0];
+  int batch_size = user_input_batch / model_input_batch;
+  bool is_batch_execution = (batch_size != 1 ? true : false);
+
+  std::vector<int> req_id(_dev_context->getDevSize());
+
+  for (int i = 0; i < _dev_context->getDevSize(); i++)
   {
-    throw std::runtime_error("Unable to create NPU request with model id (" +
-                             std::to_string(_model_id) + ")");
+    if (createNPU_request(_dev_context->getDev(i), _model_id[i], &req_id[i]))
```

Comment:
@chunseoklee I think `getDevSize()` is always 2. Thus, for a single batch, it always call `createNPU_request` twice (= getDevSize()). It means even for a single batch, there are always two calls. Is it right?

Reply:
Yes. it is always called twice. Thus, _dev_context->getDev(2) never used in the case of non-batch execution.

Reply:
Then, I prefer to have different path for single batch and multiple batch ealier though I don't know how much overhead or impaction of `createNPU_request`. But I am okay for this code.